### PR TITLE
Fix props and add feature chart factory

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

- [x] Initial fix prop warnings and errors
- [x] Move ChartFactory to hurumap-ui
- [x] Move profileLoader hooks to hurumap-ui

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/PieChart/index.js [hunk 1]
```diff
@@ -74,10 +75,9 @@ function PieChart({
     legend ||
     (data1 && data1[0].name && data1) ||
     (data2 && data2[0].name && data2);
-
   const legendProps = legendData && {
     colorScale: colorScale1,
-    data: legendData,
+    data: legendData.map(d => ({ name: d.name, label: { text: d.label } })),
```
Comment (kilemensi) [inline src/PieChart/index.js]: I think this was a mistake in `v0.1`, legend label and data label _need_ not be the same ...e.g. in comparison mode,  there a two data points with the same name but different values and hence labels. With this approach, legend will always show label from `data1` which is incorrect.

We should either not do the automatic setting of legend labels or _combine_ labels from both `data1` & `data2`

File: src/factory/ChartFactory.js [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+    statistic = {}
+  },
+  data,
+  refrence: { label: propRefrenceLabel },
+  isComparison,
+  comparisonData,
+  refrenceData,
+  profiles
+}) {
+  const key =
```
Comment (kilemensi) [inline src/factory/ChartFactory.js]: 1. They way we use the factories, each chart already has an `id` in the JSON data, we should use that as `key`.
2. If non supplied, lets just use `shortid` to generate one.

File: stories/factory.stories.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+//             return {
+//               tooltip: `${index} Comparison Data`,
+//               x: `${index} Comparison Data`,
+//               y
+//             };
+//           })}
+//         />
+//       </div>
+//     );
+//   });
```
Comment (kilemensi) [inline stories/factory.stories.js]: We should remove it then

File: src/InsightContainer/index.js [hunk 4]
```diff
@@ -35,8 +36,8 @@ const useStyles = makeStyles(({ breakpoints, variant }) => ({
     marginTop: '1rem'
   },
   highlight: {
-    height: '100%',
     width: '100%',
+    overflow: 'hidden',
```
Comment (kilemensi) [inline src/InsightContainer/index.js]: This seems to break source:
S

File: src/LineChart.js [hunk 5]
```diff
@@ -57,6 +58,7 @@ function LineChart({ data, parts, theme, ...props }) {
       <VictoryGroup>
         {groupData.map((gd, i) => (
           <VictoryGroup
+            key={JSON.stringify(gd)}
```
Comment (kilemensi) [inline src/LineChart.js]: `gd` is an array, right? `stringify`-ing it _just_ for `key` seems to be too much.

File: stories/container.stories.js [hunk 6]
```diff
@@ -94,8 +94,7 @@ storiesOf('HURUmap UI|ChartContainers/ChartContainer', module)
                   <PieChart
```
Comment (kilemensi) [inline stories/container.stories.js]: We no longer have `shareExplore` class in `ChartContainer`

**Comment by karimkawambwa** [comment]
@kilemensi thanks for the review, sorry for the large pr changes.
about the hooks, i just thought it would make things a bit cleaner but i can revert ti to the suggestion.

File: src/LineChart.js (hunk 5 above)
Comment (karimkawambwa) [reply src/LineChart.js]: @kilemensi I can't think of anything else unique here apart from index

File: src/PieChart/index.js (hunk 1 above)
Comment (karimkawambwa) [reply src/PieChart/index.js]: @kilemensi seems like issue to keep exploring I'll revert the changing so the warning exists. Lets add an issue to track it.

File: src/LineChart.js (hunk 5 above)
Comment (kilemensi) [reply src/LineChart.js]: Cool, let’s observe the performance when used in Dominion or Takwimu.... If good enough, then no problem @karimkawambwa

**Pull request merged**